File: ctf/src/test/java/com/efficios/jabberwocky/ctf/trace/ExtractedCtfTestTrace.java
package com.efficios.jabberwocky.ctf.trace;

import java.nio.file.Paths;

import org.eclipse.tracecompass.testtraces.ctf.CtfTestTrace;
import org.junit.rules.ExternalResource;

import com.efficios.jabberwocky.ctf.trace.generic.GenericCtfTrace;
import com.efficios.jabberwocky.trace.TraceInitializationException;

public class ExtractedCtfTestTrace extends ExternalResource {

    private final CtfTestTrace fTestTrace;

    private CtfTestTraceExtractor testTraceExtractor;
    private GenericCtfTrace fTrace;

    public ExtractedCtfTestTrace(CtfTestTrace testTrace) {
        fTestTrace = testTrace;
    }

    @Override
    protected void before() {
        testTraceExtractor = CtfTestTraceExtractor.extractTestTrace(fTestTrace);
        String tracePath = testTraceExtractor.getTrace().getPath();
        try {
            fTrace = new GenericCtfTrace(Paths.get(tracePath));
        } catch (TraceInitializationException e) {
            throw new IllegalArgumentException(e);
        }
    }

    @Override
    protected void after() {
        testTraceExtractor.close();
    }

    public GenericCtfTrace getTrace() {
        return fTrace;
    }

}

File: jabberwocky-lttng/src/test/kotlin/com/efficios/jabberwocky/lttng/trace/ExtractedCtfTestTrace.kt
package com.efficios.jabberwocky.lttng.trace

import com.efficios.jabberwocky.ctf.trace.generic.GenericCtfTrace
import com.efficios.jabberwocky.trace.TraceInitializationException
import org.eclipse.tracecompass.testtraces.ctf.CtfTestTrace
import org.junit.rules.ExternalResource
import java.nio.file.Paths

class ExtractedCtfTestTrace(private val testTrace: CtfTestTrace) : ExternalResource() {

    private lateinit var testTraceExtractor: CtfTestTraceExtractor
    lateinit var trace: GenericCtfTrace

    override fun before() {
        testTraceExtractor = CtfTestTraceExtractor.extractTestTrace(testTrace)
        val tracePath = testTraceExtractor.trace.getPath()
        try {
            trace = GenericCtfTrace(Paths.get(tracePath))
        } catch (e: TraceInitializationException) {
            throw IllegalArgumentException(e)
        }

    }

    override fun after() {
        testTraceExtractor.close()
    }

}
